src/ImgixTag.js

```diff
@@ -0,0 +1,157 @@
+var util = require('./util.js'),
+    targetWidths = require('./targetWidths.js');
+
+var ImgixTag = (function() {
+  function ImgixTag(el, opts) {
+    this.el = el;
+    this.settings = opts || {};
+
+    if (!this.el) {
+      throw new Error('ImgixTag must be passed a DOM element.');
+    }
+
+    if (this.el.hasAttribute('ix-initialized') && !this.settings.force) {
+      return;
+    }
+
+    this.ixPathVal = el.getAttribute('ix-path');
+    this.ixParamsVal = el.getAttribute('ix-params');
+    this.ixSrcVal = el.getAttribute('ix-src');
+    this.ixHostVal = el.getAttribute('ix-host') || imgix.config.host;
+
+    if (this.ixPathVal && !this.ixHostVal) {
+      throw new Error('You must set a value for `imgix.config.host` or specify an `ix-host` attribute to use `ix-path` and `ix-params`.');
+    }
+
+    this.baseParams = this._extractBaseParams();
+    this.baseUrl = this._buildBaseUrl();
+    this.baseUrlWithoutQuery = this.baseUrl.split('?')[0];
+
+    this.el.setAttribute('sizes', this.sizes());
+    this.el.setAttribute('srcset', this.srcset());
+
+    if (this.el.nodeName == 'IMG') {
+      this.el.setAttribute('src', this.src());
+    }
+
+    this.el.setAttribute('ix-initialized', 'ix-initialized');
+  }
+
+  ImgixTag.prototype._extractBaseParams = function() {
+    var params;
+
+    if (this.ixPathVal) {
+      params = JSON.parse(this.ixParamsVal) || {};
+    } else {
+      // If the user used `ix-src`, we have to extract the base params
+      // from that string URL.
+      var lastQuestion = this.ixSrcVal.lastIndexOf('?'),
+          paramString = this.ixSrcVal.substr(lastQuestion + 1),
+          splitParams = paramString.split('&');
+
+      params = {};
+      for (var i = 0, splitParam; i < splitParams.length; i++) {
+        splitParam = splitParams[i].split('=');
+
+        params[splitParam[0]] = splitParam[1];
+      }
+    }
+
+    // Encode any passed Base64 variant params
+    for (var key in params) {
+      if (key.substr(-2) === '64') {
+        params[key] = util.encode64(params[key]);
+      }
+    }
+
+    return params;
+  };
+
+  ImgixTag.prototype._buildBaseUrl = function() {
+    if (this.ixSrcVal) {
+      return this.ixSrcVal;
+    } else {
```

Comment:
Minor code cleanliness issue. Since we'll have bailed out of this method with the `return` statement when the conditional is truthy, the `else` clause isn't required, and can just be part of the normal method execution, e.g. 

```js
if (this.ixSrcVal) {
  return this.ixSrcVal;
}

var path = this.ixPathVal,
    protocol = 'http';

// ...
```